### ISSUE TITLE
I test the model in an NLP task.

### ISSUE THREAD

**Issue by zshy1205** [rish-16/aft-pytorch#5]
I use aft_full model，6 layers.
and I use it in init with this code:
```
self.encoder_transformer = nn.ModuleList()
for _ in range(6):
    self.encoder_transformer.append(AFTFull(max_seqlen=500, dim=512,hidden_dim=256))
```
and in forward function, I use this code:
```
for _, layer in enumerate(self.encoder_transformer):`
    x = layer(x) + x
```

Originally I used the traditional transformer, now I replaced it with this, the training loss appeared Nan，Is something wrong? and how U use the model for many layers，please help me, Thank U.

**Comment by rish-16** [comment]
Hey, thanks. I'll get into it asap. Give me a while!

**Comment by SrijanSriv211** [comment]
> I use aft_full model，6 layers. and I use it in init with this code:
> 
> ```
> self.encoder_transformer = nn.ModuleList()
> for _ in range(6):
>     self.encoder_transformer.append(AFTFull(max_seqlen=500, dim=512,hidden_dim=256))
> ```
> 
> and in forward function, I use this code:
> 
> ```
> for _, layer in enumerate(self.encoder_transformer):`
>     x = layer(x) + x
> ```
> 
> Originally I used the traditional transformer, now I replaced it with this, the training loss appeared Nan，Is something wrong? and how U use the model for many layers，please help me, Thank U.

In `AFTFull` there this line:

```
        temp = torch.exp(temp_wbias) @ torch.mul(torch.exp(K), V)
        weighted = temp / (torch.exp(temp_wbias) @ torch.exp(K))
```

The NaN might be occuring due to denominator of `weighted` being 0. To fix that you can do `weighted = temp / ((torch.exp(temp_wbias) @ torch.exp(K)) + 1e-6)`
